src/components/pie-chart.js

```diff
@@ -1,162 +1,288 @@
-import React, { useState } from "react"
+import React, { useState, useMemo, useCallback } from "react"
 
 import { ResponsivePie } from "@nivo/pie"
 
 import ArrowDropDown from "@material-ui/icons/ArrowDropDown"
 import ArrowDropUp from "@material-ui/icons/ArrowDropUp"
+import Person from "@material-ui/icons/Person"
+import { documentToReactComponents } from "@contentful/rich-text-react-renderer"
+import { BLOCKS, MARKS } from "@contentful/rich-text-types"
 import { formatToUSD } from "../components/table-utilities"
 import "./pie-chart.scss"
 
-const STAT_DESCRIPTOR = "stat_descriptor"
-const DESCRIPTION = "description"
-const CENTRAL_PROGRAMS_LABEL = "central_programs"
-const OTHER_SPENDING = "other_spending"
-
-const ALL_OUSD_SPENDING = "all_ousd_spending"
-const CENTRAL_PROGRAMS_SPENDING = "spending"
-const INCREASE = "increase"
-const DECREASE = "decrease"
-const CHANGE = "change_from_previous_year"
-
-const COLORS = [
-  "#8c564b",
-  "#1f77b4",
-  "#ff7f0e",
-  "#2ca02c",
-  "#d62728",
-  "#9467bd",
-  "#e377c2",
-  "#7f7f7f",
-  "#bcbd22",
-  "#17becf",
-]
-
-const getPercentOfTotal = data => {
-  return Math.floor(
-    (data[CENTRAL_PROGRAMS_SPENDING] / data[ALL_OUSD_SPENDING]) * 100
+function getObjValueFromPath(path) {
+  const keyRegex = "[a-zA-Z_$]\\w*"
+  const pathRegex = `^${keyRegex}(\\.${keyRegex})*$`
+  if (!path.match(pathRegex)) {
+    throw new Error(`Invalid path: ${path}`)
+  }
+  const pathArray = path.split(".")
+  return pathArray.reduce((acc, key) => acc?.[key], this)
+}
+
+const parseObject = (objectToParse, paths) => {
+  const result = {}
+  const parse = getObjValueFromPath.bind(objectToParse)
+  for (let key in paths) {
+    result[key] = parse(paths[key])
+  }
+  return result
+}
+
+const useResizeObserverRef = callback => {
+  const resizeObserver = useMemo(() => new ResizeObserver(callback), [callback])
```

Comment:
for some reason Gatsby isn't finding ResizeObserver when it does a product build. You can see it when you run `gatsby build` instead of develop. Not sure why that is. There isn't much when I Google it. Maybe because there's no browser when it's building? Including a package for backwards compatibility is the most promising search result I've found, but haven't tried it. https://www.npmjs.com/package/resize-observer-polyfill

We could also ignore the mobile problem for now.